ingest-app/src/cmr/ingest/validation/generic_document_validation.clj

```diff
@@ -0,0 +1,166 @@
+        (if (seq duplicate-concepts)
+          (let [duplicate-concept-ids (map :concept-id duplicate-concepts)
+                field-names (mapv #(string/join "." (rest (string/split % #"\."))) fields)
+                display-values (mapv str field-values)]
+            (info "Duplicate concept IDs found: " duplicate-concept-ids)
+            [(format "Values %s for fields %s must be unique for concept type %s. Duplicate concept IDs: %s"
+                     (string/join ", " display-values)
+                     (string/join ", " field-names)
+                     (str concept-type)
+                     (string/join ", " duplicate-concept-ids))])
+          []))
+      [])))
+
+(defn- validate-by-type
+  "Validates fields based on validation type. Any new validation added to the config.json
+   should be added here and implemented in the corresponding function. 
+   validation-value is here for possible future implementations but is not needed in the 
+   uniqueness validation.
+   Returns a sequence of error messages if validation fails, empty sequence otherwise."
+  [context concept validation-type fields validation-value]
+  (case validation-type
+    "unique" (validate-uniqueness context concept fields)
+    ;; Default case
+    [(str "Unknown validation type: " validation-type)]))
+
+(defn- validate-with-schema
+  "Validates a concept against the schema validations.
+   Returns a sequence of error messages if validation fails, empty sequence otherwise."
+  [context concept schema]
+  (let [validations (:Validations schema)]
+    (if validations
+      (mapcat (fn [validation]
+                (let [validation-type (:ValidationType validation)
+                      fields (:Fields validation)
+                      validation-value (:ValidationValue validation)]
+                  (validate-by-type context concept validation-type fields validation-value)))
+              validations)
+      [])))
+
+(defn- load-schema-validation
+  "Loads a single schema validation function for a concept type and version"
+  [concept-type version]
+  (try
+    (let [schema-json (generics/read-schema-config concept-type version)
+          schema (json/parse-string schema-json true)]
+      (fn [context concept]
+        (validate-with-schema context concept schema)))
+    (catch Exception e
+      (error "Error loading schema for" concept-type "version" version ":" (.getMessage e))
+      nil)))
+
+(defn- extract-concept-metadata-spec
+  "Extract metadata specification info from concept"
+  [concept]
+  (let [parsed-data (json/parse-string (:metadata concept))]
+    (when parsed-data
+      {:name (get-in parsed-data [:MetadataSpecification :Name])
+       :version (get-in parsed-data [:MetadataSpecification :Version])})))
+
+(defn- load-schema-validators
+  "Loads all schema validators for all generic concept types"
+  []
+  (info "Loading schema validation functions for all generic concept types")
```

Comment:
I think this is debug. OPS people will not be able to do much with this statement without any other details. (granted, it probably will not add that much to the budget)